Marlin/src/HAL/HAL_STM32/timers.cpp

```diff
@@ -36,58 +36,57 @@
 // Private Variables
 // ------------------------
 
-stm32_timer_t TimerHandle[NUM_HARDWARE_TIMERS];
+HardwareTimer *timer_instance[NUM_HARDWARE_TIMERS] = { NULL };
+bool timer_enabled[NUM_HARDWARE_TIMERS] = { false };
 
 // ------------------------
 // Public functions
 // ------------------------
 
-bool timers_initialized[NUM_HARDWARE_TIMERS] = { false };
-
+// 'frequency' is in Hertz
 void HAL_timer_start(const uint8_t timer_num, const uint32_t frequency) {
-
-  if (!timers_initialized[timer_num]) {
-    uint32_t step_prescaler = STEPPER_TIMER_PRESCALE - 1,
-                       temp_prescaler = TEMP_TIMER_PRESCALE - 1;
-    switch (timer_num) {
+  if (!HAL_timer_initialized(timer_num)) {
+   // According to documentation the prescale factor is computed automatically if setOverflow is in HERTZ_FORMAT
+   switch (timer_num) {
       case STEP_TIMER_NUM:
         // STEPPER TIMER - use a 32bit timer if possible
-        TimerHandle[timer_num].timer = STEP_TIMER_DEV;
-        TimerHandle[timer_num].irqHandle = Step_Handler;
-        TimerHandleInit(&TimerHandle[timer_num], (((HAL_TIMER_RATE) / step_prescaler) / frequency) - 1, step_prescaler);
-        HAL_NVIC_SetPriority(STEP_TIMER_IRQ_NAME, STEP_TIMER_IRQ_PRIO, 0);
+        timer_instance[timer_num] = new HardwareTimer(STEP_TIMER_DEV);
+        //timer_instance[timer_num]->setMode(1, TIMER_OUTPUT_COMPARE); // No pin output just the interrupt
+        //timer_instance[timer_num]->setOverflow(frequency, HERTZ_FORMAT);
```

Comment:
Ok, so at this point we're now ignoring the `frequency` parameter to initialize the interrupt. Using the PRESCALER initialization as with @MagoKimbra seems like it should be fine, assuming it's producing the exact frequency we want. But I want to be very sure this is going to match what our current timing calculations expect.